gapic/src/main/com/google/gapid/image/ArrayImage.java

```diff
@@ -541,6 +546,65 @@ public boolean isDark() {
     }
   }
 
+  /**
+   * An {@link ArrayImage} that represents an 8bit count image.
+   */
+  public static class Count8Image extends Luminance8Image {
+    public Count8Image(Image.Key key, int width, int height, int depth, byte[] data) {
+      this(width, height, depth, data,
+          getUnchecked(PIXEL_INFO_CACHE, key, () -> IntPixelInfo.compute(data, false, false)));
+    }
+
+    private Count8Image(int width, int height, int depth, byte[] data, PixelInfo info) {
+      super(width, height, depth, data, info);
+    }
+
+    @Override
+    protected Image create(int w, int h, int d, byte[] pixels) {
+      return new Count8Image(w, h, d, pixels, getInfo());
+    }
+
+    @Override
+    public Set<Stream.Channel> getChannels() {
+      return Images.COUNT_CHANNELS;
+    }
+
+    @Override
+    protected PixelValue getPixel(int x, int y) {
+      return new Pixel(data[y * width + x]);
+    }
+
+    @Override
+    public void bin(Binner binner) {
+      for (int i = 0; i < data.length; i++) {
+        binner.bin(UnsignedBytes.toInt(data[i]) / 255.0f, Stream.Channel.Count);
+      }
+    }
+
+    @Override
+    public boolean isCount() {
+      return true;
+    }
+
+    protected static class Pixel implements PixelValue {
+      private final int count;
+
+      public Pixel(byte count) {
+        this.count = count & 0xFF;
```

Comment:
Use `UnsignedBytes.toInt` instead of `& 0xFF` for readability.

Reply:
Done